docs(eslint-plugin):improve docs [consistent-type-exports] (#4792)
typescript-eslint/typescript-eslint: packages/eslint-plugin/docs/rules/consistent-type-exports.md

Given a class `Button`, and an interface `ButtonProps`, examples of code:

```ts

> export interface ButtonProps{
>   onClick: () => void;
> }
> 
> class Button implements ButtonProps {
>   onClick(){
>     console.log('button!')
>   }
> }

Reviewer: The example should illustrated re-exporting types. For example:

```suggestion
interface ButtonProps {
  onClick: () => void;
}

class Button implements ButtonProps {
  onClick() {
    console.log('button!');
  }
}

export { Button };
export type { ButtonProps };
```

Probably we should also give an invalid example, like `export { Button, ButtonProps };`

Also, I believe code style can be fixed with Prettier. Please remember to format files before committing.

Given a class `Button`, and an interface `ButtonProps`, examples of code:

```ts
interface ButtonProps {
  onClick: () => void;
}
class Button implements ButtonProps {
  onClick() {
    console.log('button!');
  }
}
export { Button };
export type { ButtonProps };

//invalid example

> export { Button, ButtonProps };

Reviewer: Please refer to how other rule docs are written. You should use two tabs, one incorrect and one correct.

Author: @Josh-Cena Should I write more explanations about the wrong example?

Given a class `Button`, and an interface `ButtonProps`, examples of code:

> ### ❌ Incorrect

Reviewer: The `<!--tabs-->` comment before the heading is actually useful. It will turn the headings into tabs. See https://typescript-eslint.io/rules/array-type#array for example.

Author: ok! I added <!--tabs--> above examples

### ✅ Correct

```ts
export { Button, type ButtonProps } from 'some-library';
```

## When Not To Use It

- If you are using a TypeScript version less than 3.8, then you will not be able to use this rule as type exports are not supported.
- If you specifically want to use both export kinds for stylistic reasons, you can disable this rule.

> - Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

Reviewer: Probably a minor rewording so it goes more along the lines of "If you use `--isolatedModules` you don't need this rule since TS would enforce type-export specifiers"

### ✅ Correct

```ts
export { Button, type ButtonProps } from 'some-library';
```

## When Not To Use It

- If you are using a TypeScript version less than 3.8, then you will not be able to use this rule as type exports are not supported.
- If you specifically want to use both export kinds for stylistic reasons, you can disable this rule.

> - If you use --isolatedModules, you don't need this rule since TS would enforce type-export specifiers.

Reviewer: Not critical, but I think it's clearer this way:

```suggestion
- If you use `--isolatedModules`, you don't need this rule since the compiler would error if a type is not re-exported using `export type`.
```

### ✅ Correct

```ts
export { Button, type ButtonProps } from 'some-library';
```

## When Not To Use It

- If you are using a TypeScript version less than 3.8, then you will not be able to use this rule as type exports are not supported.
- If you specifically want to use both export kinds for stylistic reasons, you can disable this rule.

> - If you use `--isolatedModules`, you don't need this rule since the compiler would error if a type is not re-exported using `export type`.

Reviewer 2: I imagine some teams would want this rule as a stylistic choice regardless of their `--isolatedModules` setting.

> - If you use `--isolatedModules` the compiler would error if a type is not re-exported using `export type`. If you also don't wish to enforce one style over the other, you can disable this rule.

Reviewer: Split this into two items

### ✅ Correct

```ts
export { Button, type ButtonProps } from 'some-library';
```

## When Not To Use It

- If you are using a TypeScript version less than 3.8, then you will not be able to use this rule as type exports are not supported.
- If you specifically want to use both export kinds for stylistic reasons, you can disable this rule.

> - If you use `--isolatedModules` the compiler would error if a type is not re-exported using `export type`.
> - If you also don't wish to enforce one style over the other, you can disable this rule.

Reviewer 2: @Josh-Cena my intent here was for these to be on the same line. 

```suggestion
- If you use `--isolatedModules` the compiler would error if a type is not re-exported using `export type`. If you also don't wish to enforce one style over the other, you can disable this rule.
```

Reviewer: Really...? They look like completely disjoint points though.
